### ISSUE TITLE
Download error due to name+directory length (string length)

### ISSUE THREAD

**Issue by Elpuma** [libgenapps/LibgenDesktop#26]
![error](https://user-images.githubusercontent.com/3074185/81310949-fcfe0000-9041-11ea-89bb-2abf53a61a49.jpg)
Basically the length of the directory + name is too big and it is failing, while I can easily get the address and go to the browser and get the file by downloading manually (without a length error) this would be against the automation of the LibgenDesktop app. I have been trying to fix it myself but my expertise is not that advanced and so far have not being able to fix it. I would highly appreciate a guideline to point me in the right direction so I can keep trying and I can give you all my code for your review, I know you are about to update to a new version, is this scenario covered on that version or should I keep trying? thanks in advance man!

**Comment by libgenapps** [comment]
Ah, I think you found a bug. Here's what's happening here: Windows has a limit of 260 characters for a file path. There is a way to create a file with a longer file path but then not all apps will be able to open such file. There is a check in the Libgen Desktop for such scenario: if a file path gets longer than 260 characters due to a long author + title, it replaces the file name with MD5 hash and the extension. In your case the file path is 257 characters which fits into the limit, so no substitution is needed. However it then creates a temporary file by appending `.part` to the file name which increases the file path up to 262 characters and there is no check for the length at that point.

So basically this is an edge case for file paths 256-260 characters long. I will fix this in the upcoming release which should be ready in about two days from now.

**Comment by Elpuma** [comment]
oh brother, you are awesome! thanks for all the effort, I'm still a rookie programming learner, but I'm fascinated! currently studying day and night C# just for the mere purpose to contribute in this project! I have a long list of things that I would love to add or present in a different manner, obviously biased on my own subjectivity, but I'm in love with your project!

**Comment by libgenapps** [comment]
Fixed in 1.4.0.
